tests/sps/test_figure.py

```diff
@@ -0,0 +1,181 @@
+            ]
+            }
+        ]
+        
+        self.assertEqual(extract, expected_output)
+
+
+    def test_extract_with_fig_group_and_subtag(self):
+        xml = ("""
+			<article xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:mml="http://www.w3.org/1998/Math/MathML" dtd-version="1.0" article-type="research-article" xml:lang="en">
+				<front>
+					<article-meta>
+                        <fig-group id="dogpix4">
+                            <caption><title>Figures 12-14 Bonnie Lassie</title>
+                            <p>Three perspectives on My Dog</p>
+                            </caption>
+                        <fig id="fg-12">
+                            <label>a.</label>
+                                <caption>
+                                    <title><p>View A: From the Front, Laughing</p></title>
+                                </caption>
+                                <graphic xlink:href="frontView.png"/>
+                            </fig>
+                        <fig id="fg-13">
+                            <label>b.</label>
+                            <caption>
+                                <title><p>View B: From the Side, Best Profile</p></title>
+                            </caption>
+                            <graphic xlink:href="sideView.png"/>
+                            </fig>
+                        <fig id="fg-14">
+                            <label>c.</label>
+                            <caption>
+                                <title><p>View C: In <italic>Motion</italic>, A Blur on Feet</p></title>
+                            </caption>
+                            <graphic xlink:href="motionView.png"/>
+                        </fig>
+                        </fig-group>
+					</article-meta>
+				</front>
+			</article>
+		""")
+        xml = etree.fromstring(xml)
+        extract = Figure(xml).extract_figures(subtag=True)
+        
+        expected_output = [
+            {
+			'fig_group_id': 'dogpix4', 
+			'fig_group_title': 'Figures 12-14 Bonnie Lassie', 
+			'figs': [
+				{
+				'id': 'fg-12', 
+                'title': '<p>View A: From the Front, Laughing</p>',
+				'label': 'a.', 
+				'graphic': 'frontView.png'
+                }, 
+                {
+				'id': 'fg-13',
+                'title': '<p>View B: From the Side, Best Profile</p>', 
+                'label': 'b.', 
+                'graphic': 'sideView.png'
+                }, 
+                {
+            	'id': 'fg-14',
+                'title': '<p>View C: In <italic>Motion</italic>, A Blur on Feet</p>', 
+                'label': 'c.', 
+                'graphic': 'motionView.png'
+                }
+            ]
+            }
+        ]
+        
+        self.assertEqual(extract, expected_output)
+
+
+    def test_extract_without_fig_group(self):
+        xml= xml_utils.get_xml_tree('tests/samples/0034-8910-rsp-48-2-0206.xml')
+        extract = Figure(xml).extract_figures(subtag=False)
```

Comment:
@samuelveigarangel acho que facilita não haver o `subtag` e como padrão montar o dicionário com os valores com e sem subtag, por exemplo:

```json
{
'title_without_subtag' : 'Title',
'title': '<bold>Title</bold>',
}
```
Isso porque são muitos itens retornando. Acho que o parâmetro faz sentido se for para um item.